Kernel/Bus/USB/UHCI/UHCIDescriptorPool.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2021, Jesse Buhagiar <jooster669@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/NonnullOwnPtr.h>
+#include <AK/OwnPtr.h>
+#include <AK/Stack.h>
+#include <Kernel/Memory/MemoryManager.h>
+#include <Kernel/Memory/Region.h>
+#include <Kernel/StdLib.h>
+
+namespace Kernel::USB {
+
+// NOTE: This pool is actually bounded by `PAGE_SIZE / sizeof(T)` which is the size passed
+// into the second template argument of AK::Stack. This class does NOT get to the point
+// where we will do dynamic allocation. Once you are out of descriptors, you'll receive
+// `nullptr` when you try to fetch one.
+template<typename T, size_t number_of_descriptors = 64>
+class UHCIDescriptorPool {
+
+    // Ensure that we can't get into a situation where we'll write past the page
+    // and blow up
+    static_assert(sizeof(T) * number_of_descriptors <= PAGE_SIZE);
```

Comment:
This is really confusing. First we say that the pool is bounded by `PAGE_SIZE / sizeof(T)`, but then there's also the `number_of_descriptors`.
Why do we need two separate bounds here?
Can we make `PAGE_SIZE / sizeof(T)` the only bound? Anything else would be a waste of memory anyway :)

Reply:
Sure! Having a sizeable pool within the bounds of the page (so you can have as many or as little as you want) seemed like a good idea, but it's definitely a waste of memory considering we have the whole page anyway.

Reply:
I've bound to this just `PAGE_SIZE / sizeof(T)` now. It has the added side-effect of making the template instantiation look a lot cleaner too :^)